src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java

```diff
@@ -138,4 +110,185 @@ private static Stream<Arguments> prepareData() {
             },
             false));
   }
+
+  /** Verify a scenario when structure is enriched with another field. */
+  @Test
+  public void alterStructure_noSchema() throws Exception {
+    // k1, k2
+    String testStruct1 = "{ \"testStruct\": { \"k1\" : 1, \"k2\" : 2 } }";
+    insertWithRetry(testStruct1, 0, false);
+    waitForOffset(1);
+
+    // k1, k2 + k3
+    String testStruct2 = "{ \"testStruct\": { \"k1\" : 1, \"k2\" : 2, \"k3\" : \"foo\" } }";
+    insertWithRetry(testStruct2, 1, false);
+    waitForOffset(2);
+
+    // k1, k2, k3 + k4
+    String testStruct3 =
+        "{ \"testStruct\": { \"k1\" : 1, \"k2\" : 2, \"k3\" : \"bar\", \"k4\" : 4.5 } }";
+    insertWithRetry(testStruct3, 2, false);
+    waitForOffset(3);
+
+    List<DescribeTableRow> columns = describeTable(tableName);
+    assertEquals(
+        columns.get(1).getType(),
+        "OBJECT(k1 NUMBER(19,0), k2 NUMBER(19,0), k3 VARCHAR(16777216), k4 FLOAT)");
+
+    // k2, k3, k4
+    String testStruct4 = "{ \"testStruct\": { \"k2\" : 2, \"k3\" : 3, \"k4\" : 4.34 } }";
+    insertWithRetry(testStruct4, 3, false);
+    waitForOffset(4);
+
+    columns = describeTable(tableName);
+    assertEquals(
```

Comment:
let's stick to the older tests and use assertJ with assertThat, imho is much nicer than this junit sh*t :D